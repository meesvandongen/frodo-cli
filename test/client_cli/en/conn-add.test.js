import cp from 'child_process';
import { promisify } from 'util';

const exec = promisify(cp.exec);
const CMD = 'frodo connections add --help';
const { stdout } = await exec(CMD);

test("CLI help interface for 'add' should be expected english", async () => {
  // Arrange
  expect(stdout).toMatchSnapshot();
});
