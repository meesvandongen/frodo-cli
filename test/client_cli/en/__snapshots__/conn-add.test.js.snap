// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI help interface for 'add' should be expected english 1`] = `
"Usage: frodo conn add [options] <host> [user] [password] [key] [secret]

Add a new connection profiles. You have to specify a URL, username and password at a minimum.
Optionally, for Identity Cloud, you can also add a log API key and secret.

Arguments:
  host                                         Access Management base URL, e.g.: https://cdk.iam.example.com/am. To use a connection profile, just specify a unique substring.
  user                                         Username to login with. Must be an admin user with appropriate rights to manage authentication journeys/trees.
  password                                     Password.
  key                                          API key for logging API.
  secret                                       API secret for logging API.

Options:
  -m, --type <type>                            Override auto-detected deployment type. Valid values for type: 
  classic:  A classic Access Management-only deployment with custom layout and configuration. 
  cloud:    A ForgeRock Identity Cloud environment. 
  forgeops: A ForgeOps CDK or CDM deployment. 
  The detected or provided deployment type controls certain behavior like obtaining an Identity Management admin token or not and whether to export/import referenced email templates or how to walk through the tenant admin login flow of Identity Cloud and handle MFA (choices: "classic", "cloud", "forgeops")
  -k, --insecure                               Allow insecure connections when using SSL/TLS. Has no effect when using a network proxy for https (HTTPS_PROXY=http://<host>:<port>), in that case the proxy must provide this capability. (default: Don't allow insecure connections)
  --verbose                                    Verbose output during command execution. If specified, may or may not produce additional output.
  --debug                                      Debug output during command execution. If specified, may or may not produce additional output helpful for troubleshooting.
  --curlirize                                  Output all network calls in curl format.
  --no-validate                                Do not validate connection.
  --authentication-service [service]           Name of the authentication service/tree to use.
  --authentication-header-overrides [headers]  Map of headers: {"host":"am.example.com:8081"}.
  -h, --help                                   Help
"
`;
